enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -166,26 +166,37 @@ logic, the modified script will:
    kubernetes resources, and the bootstrap etcd database snapshot to
    create a new Ignition config for the host.
 
-In order to successfully complete the installation, the user must tell
-`coreos-install` where to write the operating system image on the
-host's local storage. It may be necessary to provide other inputs to
-`coreos-install` at the same time (network addresses, etc.), and we
-have not yet completed the investigation and experimentation work
-required to design a supportable API for the complex details that may
-be needed. Therefore, to unblock progress on the bootstrap-in-place
-feature, we will temporarily treat installing the operating system as
-a manual step. Users running the OpenShift installer by hand will need
-to login to the host via `ssh` and run `coreos-install`, passing the
-appropriate arguments. Users relying on integration with Hive and the
-assisted installer for multi-cluster management will not need to
-perform this step by hand, since the agent driven by the assisted
-installer service runs `coreos-install` already. The installer and
-edge teams will work on this area further during this next release
-cycle, and use another enhancement proposal to describe the
-implementation of an API to replace the manual step.
-
-After the user runs the `coreos-install` and the RHCOS image is
-written to disk, they also need to reboot the host.
+In order to successfully complete the installation, we must know
+where to write the operating system image on the host's local
+storage. Initially this information will be passed as a single value
+in the `install-config.yaml`, using a schema like
+
+```yaml
+coreOSInstallation:
+  rootDevice: '/dev/vda'
```

Comment:
Out of curiosity, is this already covered by `.platform.baremetal.rootDeviceHints.deviceName`? Will installs at this point always use the baremetal platform?

Reply:
Single node uses None platform since baremetal platform comes with irrelevant components such as keepalived for the VIPs and haproxy.

Reply:
It's similar, so we may want to use a similar structure. But it's not a "hint" in the same way, since it's just the device name and can't accept HCTL or other less-specific values. Unless maybe we want coreos-install to support that, which would be a bigger change, and to be honest I don't think it's worth it.